tests/unit/test_application_certificate_dialog.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+from uaclient.application_certificate_dialog import ApplicationCertificateDialog
+from unittest.mock import patch, Mock
+from PyQt5.QtWidgets import QFileDialog
+
+
+@pytest.fixture
+def uaclient():
+    uaclient = Mock()
+    uaclient.application_certificate_path = "parent/certificate/path"
+    uaclient.application_private_key_path = "parent/private/key/path"
+    yield uaclient
+
+
+@pytest.fixture
+def parent(uaclient):
+    parent = Mock()
+    parent.uaclient = uaclient
+    yield parent
+
+
+def test_certificate_path(qtbot, uaclient, parent):
+    acd = ApplicationCertificateDialog(parent)
+    assert acd.ui.certificateLabel.text() == "parent/certificate/path"
+    assert acd.certificate_path == "parent/certificate/path"
+    acd.certificate_path = "None"
+    assert not acd.certificate_path
+    qtbot.addWidget = acd
+    acd.certificate_path = "test/path"
+    assert acd.ui.certificateLabel.text() == "test/path"
+
+
+def test_private_key_path(qtbot, uaclient, parent):
+    acd = ApplicationCertificateDialog(parent)
+    assert acd.ui.privateKeyLabel.text() == "parent/private/key/path"
+    assert acd.private_key_path == "parent/private/key/path"
+    acd.private_key_path = "None"
+    assert not acd.private_key_path
+    qtbot.addWidget = acd
+    acd.private_key_path = "test/path"
+    assert acd.ui.privateKeyLabel.text() == "test/path"
+
+
+def test_get_certificate(parent, uaclient):
+    acd = ApplicationCertificateDialog(parent)
+    with patch.object(QFileDialog, "getOpenFileName") as dialog:
```

Comment:
You can provide the return value right here instead of doing it below:

```suggestion
    with patch.object(QFileDialog, "getOpenFileName", return_value = "/test/path") as dialog:
```